Infrastructure: remove hidden 'Website:' label from connection dialog

SlySven opened the pull request:
This will close #6087 and address one task in #6039. ~~The latter also involved renaming the widget for the website label from `label_7` to `label_website`.~~

~~It turns out that there is no margin/space set around the elements of the informational area, so it was necessary to add an "indent" to the "Website" label to push it away from the left side. Also, by changing the alignment of the details that are shown for the website for the selected MUD to be centered in both horizontal AND vertical direction it removed the need for each entry in the: `(OrderedMap<QString, GameDetails>) mudlet::scmDefaultGames` that populates that field to include the HTML-like `<center>`...`</center>` tags to produce the same (horizontal centring) effect.~~

*It turns out that we don't really want it anyhow. With it's removal there is no need to retain the remaining two widgets in a `QGridLayout` anyhow, and they can be placed in an easier to maintain `QVBoxLayout` instead!*

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

*Edited 2022/05/27 to reflect revisions to PR*

SlySven committed:
Fix: restore Website label visibility on Connect Profile dialogue

This will close #6087 and address one task in #6039. The latter also
involved renaming the widget for the website label from `label_7` to
`label_website`.

It turns out that there is no margin/space set around the elements of the
informational area, so it was necessary to add an "indent" to the "Website"
label to push it away from the left side. Also, by changing the alignment
of the details that are shown for the website for the selected MUD to be
centered in both horizontal AND vertical direction it removed the need
for each entry in the:
`(OrderedMap<QString, GameDetails>) mudlet::scmDefaultGames`
that populates that field to include the HTML-like `<center>`...`</center>`
tags to produce the same (horizontal centering) effect.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Before:
![Screenshot 2022-05-23 010526](https://user-images.githubusercontent.com/6163092/169722438-17286ed2-6d9c-42f3-ab0d-4428461ed1dc.png)

After:
![Screenshot 2022-05-22 235639](https://user-images.githubusercontent.com/6163092/169722468-6944340a-e6a6-4b5a-988e-0f15940e7ff8.png)

vadi2 commented:
Thanks for digging up the label! Looking at it now, a s given how we just improved the connection dialog to be less "busy", I think we can get rid of that label entirely. It'll still be intuitive to know that these are website links and we'll be able to add some whitespace which will be welcome. Thoughts? @Mudlet/ui-review @demonnic

demonnic commented:
I think it looks fine without the extra `Website:` label. Unless it adds something in the accessibility arena for those using screenreaders I think it should be fine to remove.

It looks fine with it, but I hadn't even noticed it was missing until issue was put in.

SlySven commented:
Well, AFAICT, user generated profiles will not have any web-site entries, unless they are copied from a predefined one, on the other hand, the URL(s) that get displayed there might not even look like URLs (well they will be blue and underlined and clickable) - what is displayed there as text is dependent on the pseudo-HTML that is embedded in the `(QString) websiteInfo` member of the `(inline static const OrderedMap<QString, GameDetails>) mudlet::scmDefaultGames` entries; for WoTMUD I got fancy and put in two (as you can see in the example) so I think there is *some* merit in showing/introducing that content...

:information_source: If it is decided to eliminate the label then the `QGroupBox` can be simplified to have a `QVBoxLayout` (and the URLs will be centred horizontally) in one (`QLabel`) widget over the Description/User entered notes in another (`QPlainTextEdit`) one - compared to the current one that uses a `QGridLayout` with the *two* labels on the top row (after this PR as it currently is) over the Description/User entered notes widget underneath.

vadi2 commented:
Could you show a mockup of how that looks like?

Kebap commented:
> I think it looks fine without the extra `Website:` label.

I agree for this example, but WoT is very unusual in that it shows alternative names for the website link.
It even has multiple links which is also not very common.
Most profiles currently only show one raw URL there instead.
I suggest we rename "http://avalon.mud.de" etc. with "Website" or alike.
That will have the same effect as adding the label but not reduce whitespace.

vadi2 commented:
That can be done in a separate PR, not a problem. It's a bit of HTML that needs editing

Kebap commented:
I meant, close this PR, do not add that label, and do that HTML editing instead.

SlySven commented:
> ... do not add that label ...

The label is not being *added* -  it is ***already there*** but it is hidden down behind the widget displaying the "Description"/"Notes". So it will still need a PR to remove it!

vadi2 commented:
Team, we're going in circles :smile: 

Here is the PR removing the hidden 'Website' label: https://github.com/Mudlet/Mudlet/pull/6113

SlySven committed:
Infrastructure: remove hidden 'Website:' label from connection dialog

It turns out that we don't really want it anyhow. With it's removal there
is no need to retain the remaining two widgets in a `QGridLayout` anyhow,
and they can be placed in an easier to maintain `QVBoxLayout` instead!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
In https://github.com/Mudlet/Mudlet/pull/6111#issuecomment-1136757678 :
> It even has multiple links which is also not very common.

BTW "Reinos de Leyenda" has three such links, one more than WoTMUD, but they are all in the same horizontal line!

I'll take a look at editing the display of the Website URLs in a separate PR after this ***revised*** one is merged...

SlySven committed:
Add: hide label for Website details when there are none to display

This was briefly discussed between me and Vadim on Discord and felt to be
a good idea - as it leaves more space for useful other content (the MUD
description or user entered notes).

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: convert RoomID and RoomWeight widgets to be QLabels

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert "Revise: convert RoomID and RoomWeight widgets to be QLabels"

This reverts commit cfbe27e8a275d5651a82024c9556574e9f42bf2e which I
applied to the wrong branch...!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
